packages/all-packages/package.json

```diff
@@ -27,7 +27,7 @@
     "build": "tsc && node build.js",
     "clean": "rimraf lib",
     "docs": "typedoc --mode modules --module commonjs --excludeNotExported --target es5 --moduleResolution node --name JupyterLab --out ../../docs/api .",
-    "watch": "watch \"npm run build\" .. --wait 10 --filter=../../scripts/watch-filter.js"
+    "watch": "tsc-watch --onSuccess \"node build.js\""
```

Comment:
Can we continue to ignore dotfiles for vi users such as myself?

Reply:
TypeScript only watches the TS files in your project.

Reply:
I get an error running `tsc-watch` due to a `\r`: 
```bash
/usr/bin/env: ‘node\r’: No such file or directory
```

Reply:
Looks like we need to pin the version: https://github.com/gilamran/tsc-watch/issues/6

Reply:
I can confirm that it works with `tsc-watch@1.0.7`.

Puzzlingly, it looks like it is recompiling twice upon file changes? At least it is logging to output twice:
```
7:45:28 PM - File change detected. Starting incremental compilation...
7:45:28 PM - File change detected. Starting incremental compilation...
7:45:32 PM - Compilation complete. Watching for file changes.
7:45:32 PM - Compilation complete. Watching for file changes.
Hash: fa35d1f5beb69f09c741
Version: webpack 2.7.0
Time: 1253ms
             Asset     Size  Chunks                    Chunk Names
    main.bundle.js  7.67 MB       1  [emitted]  [big]  main
main.bundle.js.map  9.17 MB       1  [emitted]         main
   [6] ../packages/coreutils/lib/index.js 870 bytes {1} [built]
  [27] ../packages/application/lib/index.js 5.64 kB {1} [built]
 [451] ../packages/imageviewer-extension/lib/index.js 4.11 kB {1} [optional] [built]
 [454] ../packages/markdownviewer-extension/lib/index.js 2.37 kB {1} [optional] [built]
 [455] ../packages/notebook-extension/lib/index.js 39.8 kB {1} [optional] [built]
 [456] ../packages/pdf-extension/lib/index.js 4.96 kB {1} [optional] [built]
 [457] ../packages/running-extension/lib/index.js 1.67 kB {1} [optional] [built]
 [458] ../packages/settingeditor-extension/lib/index.js 2.63 kB {1} [optional] [built]
 [459] ../packages/shortcuts-extension/lib/index.js 3.71 kB {1} [optional] [built]
 [460] ../packages/tabmanager-extension/lib/index.js 1.77 kB {1} [optional] [built]
 [461] ../packages/terminal-extension/lib/index.js 7.3 kB {1} [optional] [built]
 [462] ../packages/theme-dark-extension/lib/index.js 767 bytes {1} [optional] [built]
 [463] ../packages/theme-light-extension/lib/index.js 771 bytes {1} [optional] [built]
 [464] ../packages/tooltip-extension/lib/index.js 5.57 kB {1} [optional] [built]
 [465] ../packages/vega2-extension/lib/index.js 6.17 kB {1} [optional] [built]
    + 1147 hidden modules
Hash: fa35d1f5beb69f09c741
Version: webpack 2.7.0
Time: 1256ms
             Asset     Size  Chunks                    Chunk Names
    main.bundle.js  7.67 MB       1  [emitted]  [big]  main
main.bundle.js.map  9.17 MB       1  [emitted]         main
   [6] ../packages/coreutils/lib/index.js 870 bytes {1} [built]
  [27] ../packages/application/lib/index.js 5.64 kB {1} [built]
 [451] ../packages/imageviewer-extension/lib/index.js 4.11 kB {1} [optional] [built]
 [454] ../packages/markdownviewer-extension/lib/index.js 2.37 kB {1} [optional] [built]
 [455] ../packages/notebook-extension/lib/index.js 39.8 kB {1} [optional] [built]
 [456] ../packages/pdf-extension/lib/index.js 4.96 kB {1} [optional] [built]
 [457] ../packages/running-extension/lib/index.js 1.67 kB {1} [optional] [built]
 [458] ../packages/settingeditor-extension/lib/index.js 2.63 kB {1} [optional] [built]
 [459] ../packages/shortcuts-extension/lib/index.js 3.71 kB {1} [optional] [built]
 [460] ../packages/tabmanager-extension/lib/index.js 1.77 kB {1} [optional] [built]
 [461] ../packages/terminal-extension/lib/index.js 7.3 kB {1} [optional] [built]
 [462] ../packages/theme-dark-extension/lib/index.js 767 bytes {1} [optional] [built]
 [463] ../packages/theme-light-extension/lib/index.js 771 bytes {1} [optional] [built]
 [464] ../packages/tooltip-extension/lib/index.js 5.57 kB {1} [optional] [built]
 [465] ../packages/vega2-extension/lib/index.js 6.17 kB {1} [optional] [built]
    + 1147 hidden modules
```

Reply:
Switched to using the `watch` library, which only copies and updates the webpack bundle for files that have actually changed.